src/middlewares/memoryFencing.ts

```diff
@@ -29,17 +30,21 @@ interface MemoryFenceOptions {
  * - Responds with the specified status code and a message indicating high memory load.
  */
 export function memoryFenceMiddleware(options?: MemoryFenceOptions): Middleware {
-  const { thresholdPercent = 80, statusCode = 503 } = options || {};
+  const { thresholdPercent = 80, statusCode = 503, memoryUsageRefreshPeriod = 100 } = options || {};
   if (thresholdPercent <= 0 || thresholdPercent >= 100) {
     throw new Error('thresholdPercent must be between 1 and 100');
   }
   const limit = v8.getHeapStatistics().heap_size_limit;
   stats.gauge('memory_heap_size_limit', limit);
-
+  let { heapUsed }: { heapUsed: number } = process.memoryUsage();
+  let lastMemoryCheck = Date.now();
   return async (ctx, next) => {
-    const { heapUsed } = process.memoryUsage();
+    // Check memory usage periodically
+    if (Date.now() - lastMemoryCheck >= memoryUsageRefreshPeriod) {
```

Comment:
can we use setInterval for this to keep updating heapUsed, they we don't see to really check the condition everytime.